src/app/api/simuler/__tests__/route.test.ts

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { NextRequest, NextResponse } from 'next/server'
+import { requestAzureClientCredentialsToken } from '@navikt/oasis'
+import * as route from '../route'
+import * as routeUtils from '../routeUtils'
+
+global.fetch = jest.fn()
+
+const originalEnv = process.env
+
+jest.mock('@navikt/oasis', () => ({
+  requestAzureClientCredentialsToken: jest.fn(),
+}))
+
+jest.mock('next/server', () => ({
+  NextRequest: jest.fn().mockImplementation((url, init) => ({
+    url,
+    method: init.method,
+    headers: init.headers,
+    json: jest.fn().mockResolvedValue(init.body),
+  })),
+  NextResponse: {
+    json: jest.fn((body, init) => ({
+      status: init.status,
+      json: jest.fn().mockResolvedValue(body),
+    })),
+  },
+}))
+
+describe('route.ts', () => {
+  afterEach(() => {
+    Object.defineProperty(process.env, 'NODE_ENV', {
+      value: originalEnv,
+      writable: true,
+    })
+    jest.clearAllMocks()
+  })
+
+  describe('generateBearerToken', () => {
+    it('Burde returnere en token når kallet på requestAzureClientCredentialsToken er vellykket', async () => {
+      ;(requestAzureClientCredentialsToken as jest.Mock).mockResolvedValue({
+        ok: true,
+        token: 'test-token',
+      })
+
+      const token = await routeUtils.generateBearerToken()
+      expect(token).toBe('test-token')
+    })
+
+    it('Burde kaste en error ved feilet token generering', async () => {
+      ;(requestAzureClientCredentialsToken as jest.Mock).mockResolvedValue({
+        ok: false,
+      })
+
+      await expect(routeUtils.generateBearerToken()).rejects.toThrow(
+        'Failed to generate token'
+      )
+    })
+  })
+
+  describe('POST', () => {
+    it('Burde kalle postDev i dev miljø', async () => {
+      process.env = {
+        ...originalEnv,
+        NODE_ENV: 'development',
+      }
+
+      const resp = await route.POST({} as unknown as NextRequest)
+      expect(resp?.status).toBe(405)
+    })
+
+    it('Burde kalle postProd i prod miljø', async () => {
+      process.env = {
+        ...originalEnv,
+        NODE_ENV: 'production',
+      }
+
+      const resp = await route.POST({} as unknown as NextRequest)
+      expect(resp?.status).toBe(405)
+    })
+  })
+
+  describe('postDev', () => {
+    it('Burde returnere "Method not allowed" for requests som ikke er POST', async () => {
```

Comment:
Den sjekken for POST er felles for postDev og postProd. Burde den løftes til POST funksjonen istedet?
Slik ville man unngå duplisering av kode og test.